devices/HapticGlove/src/HapticGlove.cpp

```diff
@@ -0,0 +1,621 @@
+    m_glove->getHandJointsAngles(gloveData.humanJointValues);
+
+    // actuators
+    for(size_t i=0; i < nFingers; i++)
+    {
+        gloveData.fingersForceFeedback[i]=gloveData.fingersHapticFeedback[i];
+        gloveData.fingersVibroTactileFeedback[i]=gloveData.fingersHapticFeedback[ i + nFingers ];
+    }
+    m_glove->setFingersForceReference(gloveData.fingersForceFeedback);
+    m_glove->setBuzzMotorsReference(gloveData.fingersVibroTactileFeedback);
+
+    return true;
+}
+
+HapticGlove::HapticGlove()
+    : PeriodicThread(period)
+    , pImpl{new SenseGloveImpl()}
+{}
+
+// Destructor
+HapticGlove::~HapticGlove() = default;
+
+bool HapticGlove::open(yarp::os::Searchable& config)
+{
+    yInfo()<<LogPrefix<<"HapticGlove::open(yarp::os::Searchable& config)";
+
+    // ==================================
+    // Check the configuration parameters
+    // ==================================
+    // Period of the this device
+    if (!(config.check("period") && config.find("period").isFloat64())) {
+        yInfo() << LogPrefix << "Using default period: " << period << "s";
+    }
+    else {
+        period = config.find("period").asFloat64();
+        yInfo() << LogPrefix << "Using the period : " << period << "s";
+    }
+
+    // Get port prefix name
+    if (!(config.check("portsPrefixName") && config.find("portsPrefixName").isString())) {
+        yInfo() << LogPrefix << "Using default port prefix /wearable/senseGlove"; // to check?
+    }
+    else {
+        pImpl->portsPrefix = config.find("portsPrefixName").asString();
+        yInfo() << LogPrefix << "Using the ports prefix " << pImpl->portsPrefix;
+    }
+
+    if (!(config.check("wearableName") && config.find("wearableName").isString())) {
+        yInfo() << LogPrefix << "Using default wearable name SenseGlove";
+        pImpl->wearableName = DeviceName;
+    }
+    else {
+        pImpl->wearableName = config.find("wearableName").asString();
+        yInfo() << LogPrefix << "Using the wearable name " << pImpl->wearableName;
+    }
+
+    // Configure the implementation class
+    if(!pImpl->configure(config))
+    {
+        yInfo() << LogPrefix << "Cannot configure the implementation class";
+        return false;
+    }
+
+    // ===================
+    // Ports configuration ???
+    // ===================
+//    if(!pImpl->dataPort.open(pImpl->portsPrefix + ":o")) {
+//        yError() << LogPrefix << "Failed to open data port " << pImpl->portsPrefix + ":o";
+//        return false;
+//    }
+
+    pImpl->linkSensorPrefix =getWearableName() +sensor::IVirtualLinkKinSensor::getPrefix();
+    pImpl->sensegloveLinkSensor= SensorPtr<SenseGloveImpl::SenseGloveVirtualLinkKinSensor>{std::make_shared<SenseGloveImpl::SenseGloveVirtualLinkKinSensor>(pImpl.get(),
+                                                                                                                                                           pImpl->linkSensorPrefix + pImpl->gloveData.humanHandLinkName)};
```

Comment:
Is it necessary to use also `SensorPtr` here beyond the `make_shared` call?

Reply:
About this point I am not sure, I tried to follow other implementations.

Reply:
Blindly, this could maybe work?

```cpp
pImpl->sensegloveLinkSensor = std::make_shared<SenseGloveImpl::SenseGloveVirtualLinkKinSensor>(
    pImpl.get(), pImpl->linkSensorPrefix + pImpl->gloveData.humanHandLinkName);
```

Reply:
As you suggested, and I agree with you, they are redundant initialization. I removed them from the related parts of the code.